task/host-sp-comms/src/bsp/gimlet_bcd.rs

```diff
@@ -0,0 +1,674 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+//! SP inventory types and implementation
+//!
+//! This reduces clutter in the main `ServerImpl` implementation
+use super::ServerImpl;
+
+use drv_i2c_api::I2cDevice;
+use drv_i2c_api::ResponseCode;
+use drv_i2c_devices::at24csw080::{At24Csw080, Error as EepromError};
+use drv_oxide_vpd::VpdError;
+use drv_spi_api::SpiServer;
+use userlib::TaskId;
+use zerocopy::AsBytes;
+
+use host_sp_messages::{InventoryData, InventoryDataResult};
+
+userlib::task_slot!(I2C, i2c_driver);
+userlib::task_slot!(SPI, spi_driver);
+
+impl ServerImpl {
+    /// Number of devices in our inventory
+    pub(crate) const INVENTORY_COUNT: u32 = 60;
+
+    /// Look up a device in our inventory, by index
+    ///
+    /// Indexes are assigned arbitrarily and may change freely with SP
+    /// revisions.
```

Comment:
We currently require a host reboot any time we're updating the SP, so this seems totally fine. However, if we ever want to support updating the SP _without_ rebooting the host, it could be problematic: if the host caches the set of device indices and uses them to (e.g.) query sensors at runtime, and the SP is updates and reorders the indices, that host cache will now be out of date. Chatted about this briefly - to support that, we probably want to add something that will allow the SP to tell if a host's request for sensor data matches its current sensor list, and return a "your cache is out of date / I've been updated" error if it does not. (To be clear, this is a problem for future us: "updating the SP requires rebooting the host" is fine!)